File: Nejman.Newtone.Core/Data/Playlists.cs
using Nejman.Newtone.Core.IO;
using Nejman.Newtone.Core.Media;
using System;
using System.Collections.Generic;
using System.Text;

namespace Nejman.Newtone.Core.Data
{
    internal static class Playlists
    {
        public static void CreateIfNotExists(string playlistName)
        {
            if(!CoreGlobal.Playlists.ContainsKey(playlistName))
            {
                CoreGlobal.Playlists.Add(playlistName, new List<string>());
            }
        }

        public static void Add(string playlistName, MediaSource track)
        {
            Add(playlistName, track.Path);
        }

        public static void Add(string playlistName, string track)
        {
            CreateIfNotExists(playlistName);

            if (CoreGlobal.Playlists[playlistName].Contains(track))
            {
                return;
            }

            CoreGlobal.Playlists[playlistName].Add(track);
            CoreGlobal.PlaylistsRefresh.OnNext(playlistName);
        }

        public static void Remove(string playlistName)
        {
            if(!CoreGlobal.Playlists.ContainsKey(playlistName))
            {
                return;
            }

            CoreGlobal.Playlists.Remove(playlistName);
            CoreGlobal.Playlists.Remove("");
            CoreGlobal.PlaylistsRefresh.OnNext(playlistName);
            DataFile.Save();
        }

        public static void Remove(string playlistName, MediaSource track)
        {
            Remove(playlistName, track.Path);
        }

        public static void Remove(string playlistName, string track)
        {
            if (!CoreGlobal.Playlists.ContainsKey(playlistName))
            {
                return;
            }

            if (!CoreGlobal.Playlists[playlistName].Contains(track))
            {
                return;
            }

            CoreGlobal.Playlists[playlistName].Remove(track);
            CoreGlobal.PlaylistsRefresh.OnNext(playlistName);
            DataFile.Save();
        }

        public static void ChangeName(string oldName, string newName)
        {
            if(!CoreGlobal.Playlists.ContainsKey(oldName))
            {
                return;
            }

            if(CoreGlobal.Playlists.ContainsKey(newName))
            {
                return;
            }

            var playlist = CoreGlobal.Playlists[oldName];
            CoreGlobal.Playlists.Add(newName, playlist);
            CoreGlobal.Playlists.Remove(oldName);
            CoreGlobal.PlaylistsRefresh.OnNext(newName);
            CoreGlobal.PlaylistsRefresh.OnNext(oldName);
            CoreGlobal.PlaylistsRefresh.OnNext("");
            DataFile.Save();
        }
    }
}

File: app/src/main/java/com/nejman/nsec/music_player/core/data/Playlists.java
package com.nejman.nsec.music_player.core.data;

import com.nejman.nsec.music_player.core.DataContainer;
import com.nejman.nsec.music_player.core.loaders.DataLoader;
import com.nejman.nsec.music_player.core.models.PlaylistModel;
import com.nejman.nsec.music_player.media.MediaSource;

import java.util.ArrayList;

public class Playlists {
    public static void createIfNotExists(String playlistName) {
        if (DataContainer.getInstance().getPlaylists().exists(playlistName)) {
            return;
        }

        DataContainer.getInstance().getPlaylists().addPlaylist(playlistName, new PlaylistModel(playlistName, null, new ArrayList<>()));
    }

    public static void add(String playlistName, MediaSource track) {
        add(playlistName, track.path);
    }

    public static void add(String playlistName, String track) {
        createIfNotExists(playlistName);

        if (DataContainer.getInstance().getPlaylists().contains(playlistName, track)) {
            return;
        }

        DataContainer.getInstance().getPlaylists().addToPlaylist(playlistName, track);
    }

    public static void remove(String playlistName) {
        if (!DataContainer.getInstance().getPlaylists().exists(playlistName)) {
            return;
        }

        DataContainer.getInstance().getPlaylists().removePlaylist(playlistName);
        try {
            DataLoader.save();
        } catch (Throwable throwable) {
            throwable.printStackTrace();
        }
    }

    public static void remove(String playlistName, MediaSource track) {
        remove(playlistName, track.path);
    }

    public static void remove(String playlistName, String filepath) {
        if (!DataContainer.getInstance().getPlaylists().exists(playlistName)) {
            return;
        }

        PlaylistModel model = DataContainer.getInstance().getPlaylists().get(playlistName);
        model.items.remove(filepath);
        DataContainer.getInstance().getPlaylists().editPlaylist(playlistName, model);
        try {
            DataLoader.save();
        } catch (Throwable throwable) {
            throwable.printStackTrace();
        }
    }

    public static void changeName(String oldName, String newName) {
        if (!DataContainer.getInstance().getPlaylists().exists(oldName)) {
            return;
        }

        if (DataContainer.getInstance().getPlaylists().exists(newName)) {
            return;
        }

        PlaylistModel model = DataContainer.getInstance().getPlaylists().get(oldName);
        model.name = newName;

        DataContainer.getInstance().getPlaylists().addPlaylist(newName, model);
        DataContainer.getInstance().getPlaylists().removePlaylist(oldName);
        try {
            DataLoader.save();
        } catch (Throwable throwable) {
            throwable.printStackTrace();
        }
    }
}
